packages/desktop/src/hooks.rs

```diff
@@ -45,14 +45,18 @@ pub fn use_wry_event_handler(
 pub fn use_muda_event_handler(
     mut handler: impl FnMut(&muda::MenuEvent) + 'static,
 ) -> WryEventHandler {
-    // move the runtime into the event handler closure
+    use dioxus_core::prelude::current_scope_id;
+
+    // Capture the current runtime and scope ID.
     let runtime = Runtime::current().unwrap();
+    let scope_id = current_scope_id().unwrap();
 
     use_wry_event_handler(move |event, _| {
```

Comment:
Instead of patching use_muda_event_handler, can you apply this patch to use_wry_event_handler so that both carry the scope with them. If the use_muda_event_handler closure has the runtime and scope, then use_wry_event_handler closure should inherit those fixes

Reply:
Sounds good 👍 that does feel better to keep the same behavior in `use_wry_event_handler`

Looks to be working in https://github.com/DioxusLabs/dioxus/pull/3120/commits/e44041a20b6f769059212ae5d7fcc923f027f840